pkg/types/kustomization.go

```diff
@@ -18,14 +18,30 @@ limitations under the License.
 package types
 
 import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"sigs.k8s.io/kustomize/pkg/patch"
 )
 
+// TypeMeta is copied from apimachinery
+type TypeMeta struct {
+	// Kind is a string value representing the REST resource this object represents.
+	// Servers may infer this from the endpoint the client submits requests to.
+	// Cannot be updated.
+	// In CamelCase.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+	// +optional
+	Kind string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
```

Comment:
Instead of copying all that, just a one line pointer
to original doc so the comments don't get out of sync
and to underscore that this field is exactly the same as the original.
```
// Kind copies apimachinery/pkg/apis/meta/v1.Typemeta.Kind
```